cocotb/decorators.py

```diff
@@ -262,7 +263,8 @@ def _force_outcome(self, outcome):
         should be inlined into `abort` below, and the call in `unschedule`
         replaced with `abort(outcome.error)`.
         """
-        assert self._outcome is None
+        if self._outcome is not None:
+            raise scheduler.InternalError("Outcome already has a value, but is being set again.")
```

Comment:
Or alternatively:
```suggestion
            # imported here to avoid circular imports
            from cocotb.scheduler import InternalError
            raise InternalError("Outcome already has a value, but is being set again.")
```